code/game/turfs/simulated/walls.dm

```diff
@@ -82,11 +86,24 @@
 	return "You can deconstruct this by welding it, and then wrenching the girder.<br>\
 			You can build a wall by using metal sheets and making a girder, then adding more metal or plasteel."
 
+/// Apply rust effects to the wall
+/turf/simulated/wall/proc/rust()
+	if(rusted)
+		return
+	name = "rusted [name]"
+	rusted = TRUE
+	update_icon()
+
 /turf/simulated/wall/proc/update_icon()
 	if(!damage_overlays[1]) //list hasn't been populated
 		generate_overlays()
 
 	QUEUE_SMOOTH(src)
+	if(!rusted_overlay)
+		rusted_overlay = icon('icons/turf/overlays.dmi', pick("rust", "rust2"), pick(NORTH, SOUTH, EAST, WEST))
+	if(rusted && rusted_overlay)
+		add_overlay(rusted_overlay)
```

Comment:
Two issues:
- This would create (but not apply) rusted overlay for *all* walls.
- This would repeatedly apply more and more (identical) rust overlays on each `update_icon()` call.

Try this:
```suggestion
	if(rusted && !rusted_overlay)
		rusted_overlay = icon('icons/turf/overlays.dmi', pick("rust", "rust2"), pick(NORTH, SOUTH, EAST, WEST))
		add_overlay(rusted_overlay)
```